rust-runtime/aws-smithy-types/src/date_time/mod.rs

```diff
@@ -0,0 +1,546 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! DateTime value for representing Smithy timestamps.
+
+use num_integer::div_mod_floor;
+use num_integer::Integer;
+use std::convert::TryFrom;
+use std::error::Error as StdError;
+use std::fmt;
+use std::time::Duration;
+use std::time::SystemTime;
+use std::time::UNIX_EPOCH;
+
+mod format;
+pub use self::format::DateTimeFormatError;
+pub use self::format::DateTimeParseError;
+
+const MILLIS_PER_SECOND: i64 = 1000;
+const NANOS_PER_MILLI: u32 = 1_000_000;
+const NANOS_PER_SECOND: i128 = 1_000_000_000;
+const NANOS_PER_SECOND_U32: u32 = 1_000_000_000;
+
+/* ANCHOR: date_time */
```

Comment:
What are these "anchors" about?

Reply:
when we generate the mdbook of smithy docs, these code snippets get included